pkg/graveler/graveler_test.go

```diff
@@ -428,6 +428,192 @@ func TestGravelerGet_Advanced(t *testing.T) {
 	}
 }
 
+func TestGraveler_Diff(t *testing.T) {
```

Comment:
You always use the commitID for the branch and the ref which makes this sort of `DiffUncommitted`. Maybe add some tests to diff with unrelated commit.